### PR TITLE
spirv-fuzz: Handle isomorphic types property in composite construction

### PR DESCRIPTION
**Pull request by afd**
The fuzzer pass that constructs composites had an issue where it would
regard isomorphic but distinct structs (similarly arrays) as being
interchangeable when constructing composites.  This change fixes the
problem by relying less on the type manager.

### CODE REVIEW

File: source/fuzz/fuzzer_pass_construct_composites.cpp [hunk 1]
```diff
@@ -164,18 +161,15 @@ void FuzzerPassConstructComposites::RecordAvailableInstruction(
 
 std::unique_ptr<std::vector<uint32_t>>
 FuzzerPassConstructComposites::TryConstructingArrayComposite(
-    const opt::analysis::Array& array_type,
+    const opt::Instruction& array_type_instruction,
```
Comment (paulthomson) [inline source/fuzz/fuzzer_pass_construct_composites.cpp:164]: This is not really part of this PR but...I find the function name confusing. Shouldn't it be more like `FindSuitableArrayComponents` or something similar? And does it need to return a unique_ptr?

Comment (afd) [reply source/fuzz/fuzzer_pass_construct_composites.cpp:164]: I addressed the name issue and made this (and the other similar functions) return a vector, not a unique_ptr, with an empty vector (rather than nullptr) representing failure.
